### PR TITLE
Islands page

### LINKED ISSUE #73: Add Islands Page
## Problem
As a user, I want the ability to view my owned islands and select a different one

## Definition of Done
- Island Page is added
- Dynamic generation of Island page with Island display template
- See which is selected
- Set which is selected

## Subtasks
- [ ] Research
- [ ] Create island page with dynamically generated island displays
- [ ] Implement functionality to update databases and auth service when user selects a different pet
- [ ] Documentation and Testing

### PR DESCRIPTION
**Pull request by wesleypayton**
Fixes #73 

## Problem
As a user, I need an island page to select which island I want to display on the timer

### Definition of Done
- User can select island from islands page
- When selected, island is assigned to auth service's CurrentIsland properties
- Same with mobile db and CurrentIslandId
- same with sync on server side and CurrentIslandId

## Solution
- Added islands page
- Added methods for localcontext and api to saved user's new selected island
- Added method to fetch user's owned islands
- User's new selected island is updated across local and api db as well as auth service

## Documentation
Documentation can be found here: https://github.com/ChrispyPeaches/FocusFriends/wiki/App-Areas#social

## How was this tested?
Tested locally

### CODE REVIEW

File: src/FocusApp.Client/Views/Social/IslandsPage.cs [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+                UserId = _authenticationService.CurrentUser.Id,
+                selectedIslandId = _authenticationService.SelectedIsland != null ? _authenticationService.SelectedIsland.Id : null
+            },
+            default);
+
+            userIslands = result.Islands;
+        }
+        catch (Exception ex)
+        {
+            throw new Exception("Error when fetching islands from local DB.", ex);
```
Comment (ChrispyPeaches) [inline src/FocusApp.Client/Views/Social/IslandsPage.cs:174]: Can we change this to be logging? I think throwing an uncaught exception might crash the app

Comment (ChrispyPeaches) [reply src/FocusApp.Client/Views/Social/IslandsPage.cs:174]: If this happens on the pets and decor pages, then we need to do the same there as well

**Review by ChrispyPeaches — CHANGES_REQUESTED** [review]
Overall really good! Just note the rquested changes

File: src/FocusApp.Client/Views/Social/IslandsPage.cs (hunk 1 above)
Comment (wesleypayton) [reply src/FocusApp.Client/Views/Social/IslandsPage.cs:174]: on it!

File: src/FocusApp.Client/Methods/User/EditUserSelectedIsland.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                // Update the user on the server
+                await _client.EditUserSelectedIsland(command, cancellationToken);
+
+                // Update the local database to reflect the changes made to the user
+                Shared.Models.User? user = await _localContext.Users.FirstOrDefaultAsync(u => u.Id == command.UserId, cancellationToken);
+
+                // Fetch island from local db if not null
+                if (command.IslandId != null)
+                {
+                    Island? island = await _localContext.Islands.Where(i => command.IslandId == i.Id).FirstOrDefaultAsync(cancellationToken);
```
Comment (iggy808) [inline src/FocusApp.Client/Methods/User/EditUserSelectedIsland.cs:47]: No need to delay PR for this, but just a note for future development: the FirstOrDefaultAsync function can take in a lambda expression that can negate the need for the Where clause!

Ex./ _localContext.Islands.FirstOrDefaultAsync(i => command.IslandId == i.Id, cancellationToken) is an equivalent statement to what you have written.

No need to make this adjustment, just wanted to shout it out!

**Review by iggy808 — APPROVED** [review]
Very nice! Approved🔥🔥

**Review by ChrispyPeaches — APPROVED** [review]
Nice work! You're killin it!

**Pull request merged**